### ISSUE TITLE
Add contextual parsing support for Clojure?

### ISSUE THREAD

**Issue by Qwerp-Derp** [CoenraadS/BracketPair#23]
This may apply to other Lisp-like languages as well, but Clojure has two types of comments:

```clojure
; This is a single-line comment, which is
; marked by a ";"

#_(
    This is a multi-line comment, which is denoted
    by "#_", then a pair of brackets. Everything between
    the brackets is ignored.
)
```

**Comment by CoenraadS** [comment]
Hello

I pushed basic support for the single line comments, but currently my comment/string parser is very simple, and it will break if a clojure multiline comment contains a bracket.
e.g. 

```
(comment


(defn hey []
  ("Hey there!"))

Check me out!
)
```

I may have another look at improving my parser to support bracket counting to support these type of comments, but this is the current situation.

**Comment by Qwerp-Derp** [comment]
I think the inside brackets in a multiline comment have to match as well, otherwise the comment thing will get confused.

**Comment by Qwerp-Derp** [comment]
Can you also add support for strings?

    "This is a string"
    "This is also a string"

Also, some things need to be ignored, such as escape sequences:

    \( \) \[ \] \{ \}

**Comment by CoenraadS** [comment]
I added the strings, I meant to do that last time but forgot. Oops.

Could you give a real world example of how escaped brackets might be used in clojure?

**Comment by CoenraadS** [comment]
This is now waiting for: https://github.com/PrismJS/prism/pull/1214

**Comment by tkjone** [comment]
I just posted #1311 which implements the updates requested for #1214.

**Comment by CoenraadS** [comment]
Fixed in 1.0.19
